docs/code/IDataViewTypeSystem.md

```diff
@@ -0,0 +1,844 @@
+* Note that for vector types, this requires the dimensionality information to
+* be identical.
+
+* The `SameSizeAndItemType` method is the same as `Equals` for non-vector
+types. For vector types, it returns true iff the two types have the same item
+type and have  the same `VectorSize` values. For example, for the two vector
+types `V<R4,3,2>` and `V<R4,6>`, `Equals` returns false but
+`SameSizeAndItemType` returns true.
+
+### `PrimitiveType` Abstract Class
+
+The `PrimitiveType` abstract class derives from `ColumnType` and is the base
+class of all primitive type implementations.
+
+### `TextType` Sealed Class
+
+The `TextType` sealed class derives from `PrimitiveType` and is a singleton-
+class for the standard text type. The instance is exposed by the static
+`TextType.Instance` property.
+
+### `BooleanType` Sealed Class
+
+The `BooleanType` sealed class derives from `PrimitiveType` and is a
+singleton-class for the standard boolean type. The instance is exposed by the
+static `BooleanType.Instance` property.
+
+### `NumberType` Sealed Class
+
+The `NumberType` sealed class derives from `PrimitiveType` and exposes single
+instances of each of the standard numeric types, `R4`, `R8`, `I1`, `I2`, `I4`,
+`I8`, `U1`, `U2`, `U4`, `U8`, and `UG`.
+
+### `DateTimeType` Sealed Class
+
+The `DateTimeType` sealed class derives from `PrimitiveType` and is a
+singleton-class for the standard datetime type. The instance is exposed by the
+static `DateTimeType.Instance` property.
+
+### `DateTimeZoneType` Sealed Class
+
+The `DateTimeZoneType` sealed class derives from `PrimitiveType` and is a
+singleton-class for the standard datetime timezone type. The instance is
+exposed by the static `DateTimeType.Instance` property.
+
+### `TimeSpanType` Sealed Class
+
+The `TimeSpanType` sealed class derives from `PrimitiveType` and is a
+singleton-class for the standard datetime timezone type. The instance is
+exposed by the static `TimeSpanType.Instance` property.
+
+### `KeyType` Sealed Class
+
+The `KeyType` sealed class derives from `PrimitiveType` and instances
+represent key types.
+
+Notes:
+
+* Two key types are considered equal iff their kind, min, count, and
+contiguous values are the same.
+
+* The static `IsValidDataKind` method returns true iff kind is `U1`, `U2`,
+`U4`, or `U8`. These are the only valid underlying data kinds for key types.
+
+* The inherited `KeyCount` property returns the same value as the `Count`
+property.
+
+### `VectorType` Sealed Class
+
+The `VectorType` sealed class derives from `ColumnType` and instances
+represent vector types. The item type is specified as the first parameter to
+each constructor and the dimension information is inferred from the additional
+parameters.
+
+* The `DimCount` property indicates the number of dimensions and the `GetDim`
+method returns a particular dimension value. All dimension values are non-
+negative integers. A zero dimension value indicates unknown (or variable) in
```

Comment:
Severely minor: recommend, "A zero-dimension value "

Reply:
I think perhaps this is a misparse. It's "zero (dimension value)". I can rephrase though, since "2-dimension" and such things are common constructs, and it does resemble that...

---
In reply to: [188812830](https://github.com/dotnet/machinelearning/pull/173#discussion_r188812830) [](ancestors = 188812830)